--- a/src/panfrost/lib/kmod/kbase_kmod.c
+++ b/src/panfrost/lib/kmod/kbase_kmod.c
@@ -1729,21 +1729,8 @@
                cpu[8], cpu[9], cpu[10], cpu[11],
                cpu[12], cpu[13], cpu[14], cpu[15]);
 
-      fprintf(stderr,
-              "PANVKDBG SHM[%u] GPU0 "
-              "%02x %02x %02x %02x "
-              "%02x %02x %02x %02x "
-              "%02x %02x %02x %02x "
-              "%02x %02x %02x %02x\n",
-              i,
-              gpu[0], gpu[1], gpu[2], gpu[3],
-              gpu[4], gpu[5], gpu[6], gpu[7],
-              gpu[8], gpu[9], gpu[10], gpu[11],
-              gpu[12], gpu[13], gpu[14], gpu[15]);
-
       if (center + 16 <= size) {
          cpu += center;
-         gpu += center;
 
          fprintf(stderr,
                  "PANVKDBG SHM[%u] CPUC "
@@ -1756,18 +1743,12 @@
                  cpu[8], cpu[9], cpu[10], cpu[11],
                  cpu[12], cpu[13], cpu[14], cpu[15]);
-
-         fprintf(stderr,
-                 "PANVKDBG SHM[%u] GPUC "
-                 "%02x %02x %02x %02x "
-                 "%02x %02x %02x %02x "
-                 "%02x %02x %02x %02x "
-                 "%02x %02x %02x %02x\n",
-                 i,
-                 gpu[0], gpu[1], gpu[2], gpu[3],
-                 gpu[4], gpu[5], gpu[6], gpu[7],
-                 gpu[8], gpu[9], gpu[10], gpu[11],
-                 gpu[12], gpu[13], gpu[14], gpu[15]);
       }
    }
 }
+@@ -2244,14 +2235,10 @@
+          .type = sync->type == PAN_KMOD_BO_SYNC_CPU_CACHE_FLUSH
+                     ? BASE_SYNCSET_OP_MSYNC
+                     : BASE_SYNCSET_OP_CSYNC,
+       };
+-<<<<<<< HEAD
+       if (getenv("PANVK_G57_DEBUG")) {
+          fprintf(stderr, "PANVKDBG mem_sync: type=%d va=%llx start=%llx size=%llu\n",
+                  (int)req.type, (unsigned long long)req.handle,
+                  (unsigned long long)sync->start, (unsigned long long)sync->size);
+       }
+-=======
+-      fprintf(stderr, "PANVKDBG mem_sync: type=%d va=%llx start=%llx size=%llu\n",
+-              (int)req.type, (unsigned long long)req.handle,
+-              (unsigned long long)sync->start, (unsigned long long)sync->size);
+->>>>>>> 5cfe37d1ac0e3d267f9661e47013e6de7ddb352f
+
+       if (pan_kmod_ioctl(dev->fd, KBASE_IOCTL_MEM_SYNC, &req)) {
+          mesa_loge("kbase: KBASE_IOCTL_MEM_SYNC failed: %s", strerror(errno));
+          return -1;
+       }
+    }
+    return 0;
+ }
